pkg/api/query.go

```diff
@@ -22,6 +24,24 @@ func getBodyQueryRunnerHandler(caller string, querier query.Runner) func(context
 	}
 }
 
+func getSSEBodyQueryRunnerHandler(caller string, querier *distributed.QueryRunner) func(context.Context, *ArgsInput, sse.Sender) {
+	return func(ctx context.Context, input *ArgsInput, send sse.Sender) {
+		querier.SetResultReceivedFn(func(res *results.Result) error {
+			if res == nil {
+				return nil
+			}
+			return send.Data(&PartialResult{res})
+		})
+
+		res, err := runQuery(ctx, caller, input.Body, querier)
+		if err != nil {
+			send.Data(err)
```

Comment:
The linter should complain about the unhandled error here I guess. Maybe `_ = send.Data(err)` as below?